fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,281 @@
+
+    This function takes in a :class:`fairlearn.metrics.MetricFrame` with precomputed metrics
+    and metric errors and a `errors` or `conf_intervals` array to interpret the columns
+    of the :class:`fairlearn.metrics.MetricFrame`.
+
+    The items at each index of the given `metrics` array and given `errors` or `conf_intervals`
+    array should correspond to a pair of the same metric and metric error, respectively.
+
+    Parameters
+    ----------
+    metric_frame : fairlearn.metrics.MetricFrame
+        The collection of disaggregated metric values, along with the metric errors.
+
+    kind : str, default="scatter"
+        The type of plot to display. e.g. "bar", "line", etc.
+        List of options is detailed in :meth:`pandas.DataFrame.plot`
+
+    metrics : str or list of str
+        The name of the metrics to plot.
+        Should match columns from the given :class:`fairlearn.metrics.MetricFrame`.
+
+    errors : str or list of str
+        The name of the error metrics to plot.
+        Should match columns from the given :class:`fairlearn.metrics.MetricFrame`.
+        Errors quantify the bounds relative to the base metric.
+        Errors must be non-negative.
+
+        Example:
+            If the error for a certain column is :code:`0.01`
+            and the metric is 0.6,
+            then the plotted bounds will be :code:`[0.59, 0.61]`
+
+        Note:
+            The return of the error function should be an scalar of the
+            symmetric errors. e.g. :code:`0.01`
+
+    conf_intervals : str or list of str
+        The name of the confidence intervals to plot.
+        Should match columns from the given :class:`fairlearn.metrics.MetricFrame`.
+
+        Note:
+            The return of the error function should be an array of the lower
+            and upper bounds. e.g. :code:`[0.59, 0.62]`
+
+    subplots : bool, default=True
+        Whether or not to plot metrics on separate subplots
+
+    plot_error_labels : bool, default=False
+        Whether or not to plot numerical labels for the error bounds
+
+    error_labels_precision : int, default=4
+        The number of digits of precision to show for error labels
+
+    error_labels_fontsize : int, default=8
+        The font size to use for error labels
+
+    error_labels_color : str, default="black"
+        The font color to use for error labels
+
+    error_labels_ha : str, default="center"
+        The horizontal alignment modifier to use for error labels
+
+    Returns
+    -------
+    :class:`matplotlib.axes.Axes` or :class:`numpy.ndarray` of them
+    """
+    _check_for_ambiguous_error_metric(errors, conf_intervals)
+    _check_for_valid_metrics_format(metrics)
+
+    metrics = [metrics] if isinstance(metrics, str) else metrics
+    conf_intervals = [conf_intervals] if isinstance(conf_intervals, str) else conf_intervals
+    errors = [errors] if isinstance(errors, str) else errors
+
+    df = metric_frame.by_group
```

Comment:
is `by_group` always defined? Should there be any validation on the given `MetricFrame`?

Reply:
I'm not too familiar if `MetricFrame` is guaranteed to have `by_group`. I can check if the user input a MetricFrame type though. Would this be sufficient?

Reply:
@MiroDudik and @riedgar-ms  would know better here

Reply:
`by_group` should always be defined. It's calculated in the constructor of `MetricFrame`